algorithmiaio/langpacks
# Algo-XXX automatic environment testing

This PR adds automatic environment testing to the packageset validator. It also adds an optional algorithmia API key environment variable that can be passed into the runtime container for algorithmia client API support.

I need feedback on:
Testing system (files, format, etc)
Structure and format (everything makes sense? sanity check)

Once I have that, I'll move forward on adding test support to all algorithm language templates & dependency templates.

TO TEST:
install new dependency pycurl (and any system dependencies required)
Run the following:
`./tools/environment_validator.py -g python3 -s python37 -t language -n python3 -a 1`

This should return a success as it validates two test cases
